### ISSUE TITLE
Running webpackDevServer doesn't hot reload the app

### ISSUE THREAD

**Issue by galex** [cfnz/muirwik-starterapp#1]
Hello,

After opening the starterapp in IntelliJ and running the task "webpackDevServer" and making some code changes, I have to stop, build and then webpackDevServer again so that I can see my changes. 

Is that the expected flow of development?

Thanks,
Alex

**Comment by galex** [comment]
I just realised it does work, I need to build the app manually.
Is there a way to build the app as soon as I finish editing?

**Comment by cfnz** [comment]
Yip, I build manually.
I actually started out with create-react-kotlin-app which does rebuild more or less continuously, so it can be configured. However, as I got into building the component wrappers, it started to take around 12-15 seconds to compile... and it got quite annoying when editing, then editing some more... the app would reload, then reload again and you were waiting longer and longer... I actually wanted it to rebuild on command to avoid this.

The compile time was also what got me started on trying to separate out the component wrappers from the app so the app would compile faster without needing to compile the wrappers.

Now with incremental compilation improvements, it might be possible again... though to be ideal it would be good to see if the react hot module reloader would work better, and also then I think would need to get into better state management for when the app reloads, it gets all its state back (i.e. so does not jump around in the UI... e.g. editing the test app and recompiling you will see that the web app jumps back to the first page - would be nice to reload state and stay on the "page" you are on).

So it could do with some improvements in this regard... not sure if it could get as fast as the videos you see of coding React apps directly, but I do like using Kotlin and a type safe language though as a trade-off.

Colin
